tachyon/math/polynomials/univariate/univariate_polynomial.h

```diff
@@ -122,10 +124,33 @@ class UnivariatePolynomial final
   template <typename Container>
   constexpr static Field EvaluateVanishingPolyByRoots(const Container& roots,
                                                       const Field& point) {
-    return std::accumulate(roots.begin(), roots.end(), Field::One(),
-                           [point](Field& acc, const Field& root) {
-                             return acc *= (point - root);
-                           });
+#if defined(TACHYON_HAS_OPENMP)
+    size_t thread_nums = static_cast<size_t>(omp_get_max_threads());
+#else
+    size_t thread_nums = 1;
+#endif
+    size_t n = roots.size();
+    if (n * 2 < thread_nums) {
```

Comment:
Please change it to something like `MultiplicativeGroup::kParallelBatchInverseDivisorThreshold` ?